main/pipeline/pipeline.cc

```diff
@@ -137,6 +140,240 @@ unique_ptr<parser::Node> runParser(core::GlobalState &gs, core::FileRef file, co
     return nodes;
 }
 
+core::LocOffsets locOffset(pm_location_t *loc, pm_parser_t *parser) {
+    uint32_t locStart = static_cast<uint32_t>(loc->start - parser->start);
+    uint32_t locEnd = static_cast<uint32_t>(loc->end - parser->start);
+
+    std::cout << "locStart: " << locStart << std::endl;
+    std::cout << "locEnd: " << locEnd << std::endl;
+
+    return core::LocOffsets{locStart, locEnd};
+}
+
+unique_ptr<parser::Node> convertPrismToSorbet(pm_node_t *node, pm_parser_t *parser, core::GlobalState &gs) {
+    switch (PM_NODE_TYPE(node)) {
+        case PM_FLOAT_NODE: {
+            pm_float_node *floatNode = (pm_float_node *)node;
+            pm_location_t *loc = &floatNode->base.location;
+
+            return make_unique<parser::Float>(locOffset(loc, parser), std::to_string(floatNode->value));
+
+            break;
+        }
+        case PM_INTEGER_NODE: {
+            pm_integer_node *intNode = (pm_integer_node *)node;
+            pm_location_t *loc = &intNode->base.location;
+
+            return make_unique<parser::Integer>(locOffset(loc, parser), std::to_string(intNode->value.value));
+
+            break;
+        }
+        case PM_PROGRAM_NODE: {
+            pm_statements_node *stmts = ((pm_program_node *)node)->statements;
+            return convertPrismToSorbet((pm_node *)stmts, parser, gs);
+
+            break;
+        }
+        case PM_RATIONAL_NODE: {
+            pm_rational_node *rationalNode = (pm_rational_node *)node;
+            pm_location_t *loc = &rationalNode->base.location;
+
+            const uint8_t *start = rationalNode->numeric->location.start;
+            const uint8_t *end = rationalNode->numeric->location.end;
+
+            std::string value = std::string(reinterpret_cast<const char *>(start), end - start);
+
+            return make_unique<parser::Rational>(locOffset(loc, parser), value);
+
+            break;
+        }
+        case PM_STATEMENTS_NODE: {
+            pm_node_list body = ((pm_statements_node *)node)->body;
+            pm_node *first = body.nodes[0];
+
+            return convertPrismToSorbet(first, parser, gs);
+
+            break;
+        }
+        case PM_STRING_NODE: {
+            pm_string_node *strNode = (pm_string_node *)node;
+            pm_location_t *loc = &strNode->base.location;
+
+            auto unescaped = strNode->unescaped;
+            auto source = std::string(reinterpret_cast<const char *>(unescaped.source), unescaped.length);
```

Comment:
We'll want to use `pm_string_source` and `pm_string_length` here, as it might depend on the type of string.